# pyTSCAN

Single cell RNA sequencing pseudotime analysis R package implemented with python.

**!! This repository is only for coding practice and concept learning. If it offends authors' legal rights, it will be closed.**

### Requirements

* Windows
* python >= 3.6.0
* numpy >= 1.15.2
* pandas >= 0.22.0
* sklearn >= 0.19.1
* scipy >= 1.1.0
* matplotlib >= 2.0.0

### Installation and Run the example

Download the package and run the code below.

```
/path/to/python3 test.py
```

The test results will be at `./test_result`

### Introduction

This repository is kind of repetition of the original paper which is published on ***Nucleic Acid Research*** in 2016. The original R package github page is https://github.com/zji90/TSCAN.

We only implemented part of the functions of this package manually. As a consequence,  we suggest only considering it as a practice instead of using it for study.

If there are any questions, please contact us.

Email : mxgaoxmu@gmail.com

### References

[1]  Ji, Z. and Ji, H., TSCAN: Pseudo-time reconstruction and evaluation in
single-cell RNA-seq analysis. ***Nucleic Acids Research*** 2016. 44: e117. 

[2]  R. Cannoodt et al., Computational methods for trajectory inference from
single-cell transcriptomics . ***European Journal of Immunology*** 2016. 46: 2496.